hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/acl/OMBucketAddAclRequest.java

```diff
@@ -116,7 +126,11 @@ OMClientResponse onFailure(OMResponse.Builder omResponse,
 
   @Override
   void onComplete(boolean operationResult, IOException exception,
-      OMMetrics omMetrics) {
+      OMMetrics omMetrics, AuditLogger auditLogger,
+      Map<String, String> auditMap) {
+    auditLog(auditLogger, buildAuditMessage(OMAction.ADD_ACL, auditMap,
+        exception, getOmRequest().getUserInfo()));
```

Comment:
`buildAuditMessage` determines success/failure based on whether `exception` is `null`.  But here we can also have failure without exception, indicated by `operationResult = false`.  I think audit log should indicate failure in this case.

https://github.com/apache/ozone/blob/c3f6b4b96a5c50834df223d36d2fe691da0e67ee/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/acl/OMBucketAddAclRequest.java#L124-L126

Reply:
@adoroszlai , thanks for the review and feedback.  Basically, I think  ACL error handling with both Result and Exception is complex.  I fired HDDS-4650m,  ACL related error handling refactor, to cleanly resolve this issue.

Reply:
Sorry for bad formatting, github on Phone is funny :)

Basically, I am suggesting we update exception=null in the else block after you have logged log.error().
Then call the auditlog() at the end of the method.

Reply:
@dineshchitlangia wouldn't `exception=null` have the effect of always logging `SUCCESS` in audit?

Reply:
Yikes, I intended to say that we set a not null value to exception and accordingly achieve it. Once I get to my laptop I will update here with proper suggestion. Thanks for catching it.

Reply:
```suggestion
    if (operationResult) {
      LOG.debug("Add acl: {} to path: {} success!", getAcls(), getPath());
    } else {
      omMetrics.incNumBucketUpdateFails();
      if (exception == null) {
        LOG.error("Add acl {} to path {} failed, because acl already exist",
            getAcls(), getPath());
        exception = new ACLAlreadyExistsException(); //we can create this new type of exception
      } else {
        LOG.error("Add acl {} to path {} failed!", getAcls(), getPath(),
            exception);
      }
    }
    auditLog(auditLogger, buildAuditMessage(OMAction.ADD_ACL, auditMap,
        exception, getOmRequest().getUserInfo()));
  }
     
```

Reply:
Hi @dineshchitlangia, I have the same idea to throw the ACLAlreadyExistsException exception when add an acl and ACL does not exist exception when remove an acl.   I plan to put the logic in ozoneAclUtil.java so that we don't need to do the exception == null check in every ACL related request.  So currently,  there are OzoneAclUtils.java which used by bucket, key and prefix ACL operation,  and OmOzoneAclMap.java which used by volume. I'd like to unify them and do some refactor which will be quite a big patch.  Will do it in HDDS-4650.

Reply:
Sound good to me. +1